cloudplatform/cloudplatform-connectivity-scp-cf/src/test/java/com/sap/cloud/sdk/cloudplatform/connectivity/ScpCfDestinationLoaderCacheTest.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright (c) 2023 SAP SE or an SAP affiliate company. All rights reserved.
+ */
+
+package com.sap.cloud.sdk.cloudplatform.connectivity;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatCode;
+
+import java.time.Duration;
+
+import org.junit.After;
+import org.junit.Test;
+
+import com.github.benmanes.caffeine.cache.Cache;
+import com.sap.cloud.sdk.cloudplatform.cache.CacheKey;
+import com.sap.cloud.sdk.cloudplatform.resilience.CacheExpirationStrategy;
+
+import io.vavr.control.Option;
+
+public class ScpCfDestinationLoaderCacheTest
+{
+    @After
+    public void resetCache()
+    {
+        ScpCfDestinationLoader.Cache.reset();
+    }
+
+    @Test
+    public void testDisable()
+    {
+        assertThat(ScpCfDestinationLoader.Cache.isEnabled()).isTrue();
+
+        // sut
+        ScpCfDestinationLoader.Cache.disable();
+        assertThat(ScpCfDestinationLoader.Cache.isEnabled()).isFalse();
+        assertThatCode(ScpCfDestinationLoader.Cache::enableChangeDetection).isInstanceOf(IllegalStateException.class);
+        assertThatCode(ScpCfDestinationLoader.Cache::instanceSingle).isInstanceOf(IllegalStateException.class);
+        assertThatCode(ScpCfDestinationLoader.Cache::instanceAll).isInstanceOf(IllegalStateException.class);
+    }
+
+    @Test
+    public void testReset()
+    {
+        final Cache<CacheKey, ?> cacheSingle = ScpCfDestinationLoader.Cache.instanceSingle();
+        final Cache<CacheKey, ?> cacheAll = ScpCfDestinationLoader.Cache.instanceAll();
+
+        // sut
+        ScpCfDestinationLoader.Cache.reset();
+        assertThat(cacheSingle).isNotSameAs(ScpCfDestinationLoader.Cache.instanceSingle());
+        assertThat(cacheAll).isNotSameAs(ScpCfDestinationLoader.Cache.instanceSingle());
```

Comment:
```suggestion
        assertThat(cacheAll).isNotSameAs(ScpCfDestinationLoader.Cache.instanceAll());
```
copy / pasta mistake I assume? also applies to the other tests below